src/physics/base/PhysicsStepUtils.i.hh

```diff
@@ -222,6 +228,9 @@ select_process_and_model(const ParticleTrackView& particle,
     // Nonzero MFP to interaction -- no interaction model
     CELER_EXPECT(physics.interaction_mfp() <= 0);
 
+    if (physics.macro_xs() == 0)
+        return {};
```

Comment:
There are a lot of new edge cases in this update (I assume because of situations you encountered while actually transporting) but I don't see any new tests in PhysicsStepUtils. Is it possible for you to reproduce these in unit tests? 😬

Reply:
Good point, I'll see what I can put together.

Reply:
This particular edge case was a side effect of an error in the discrete interaction sampling -- I was only resetting the mean free path in post-processing if a discrete interaction occurred, but it also needs to be reset if a process was sampled but the discrete interaction was rejected in the integral approach. Fixed that and removed this.